api/instance.go

```diff
@@ -82,15 +91,15 @@ type InstancePatch struct {
 	// Value is assigned from the jwt subject field passed by the client.
 	UpdaterID int
 
+	// Related fields
+	DataSourceList []*DataSourcePatch `jsonapi:"relation,dataSourceList"`
```

Comment:
This might not work.

Also it's quite rare to patch multiple items in a single request. For patch, probably to have a separate /database/<<dbid>>/datasource endpoint